### PR TITLE
Rework content fragment rendering for model

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

After the recent refactoring (see https://github.com/dust-tt/dust/pull/5835) of the supported types for chat, this PR revisits the existing logic of `renderConversationForModelMultiActions` to leverage the new additions.

### Key points to note:
- All user messages will be sent as either `content: [{type: "text", text: string}]` or as `content: [{type: "image_url", ...}]`.
- Content fragments are now grouped within the user message content array, formatted as follows:
  ```json
  [
    {
      "value": {
        "conversation": {
          "messages": [
            {
              "role": "user",
              "name": "<name>",
              "content": [
                {
                  "type": "text",
                  "text": "<attachment type=\"application/pdf\" title=\"c0d5b319-43bf-49dd-93d5-f3a1557876ce.pdf\">blob</attachment>\n"
                },
                { "type": "text", "text": "@gpt4 what's this?" }
              ]
            }
          ]
        },
        "specifications": [],
        "prompt": "<prompt>"
      },
      "error": null,
      "meta": null
    }
  ]
  ```

These changes are necessary to support vision capabilities. This PR also introduces the required types for image content fragments.

## Risk

There is a risk associated with these changes, but it is safe to rollback if needed.

### CODE REVIEW

File: front/lib/api/assistant/generation.ts [hunk 1]
```diff
@@ -123,7 +125,12 @@ export async function renderConversationForModelMultiActions({
       messages.push({
         role: "user" as const,
         name: m.context.fullName || m.context.username,
-        content,
+        content: [
```
Comment (fontanierh) [inline front/lib/api/assistant/generation.ts:128]: Might break tables query dust app (and maybe other dust apps ?)

We might want to do a pass to ensure everything still works fine there

**Review by fontanierh — APPROVED** [review]
Code LGTM. Can we mix "complex" context with regular simple string content ? If so we should be good otherwise we might need to switch dust apps to complex content as well

File: front/lib/api/assistant/generation.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/api/assistant/generation.ts:128]: Yes the array of messages can mix both complex type with regular stream: https://github.com/dust-tt/dust/pull/5835.

**Review by spolu — COMMENTED** [review]
:+1: :pray:

**Comment by flvndvd** [comment]
Tested on front-edge, works properly.

**Pull request merged**